### ISSUE TITLE
"A non-numeric value encountered" [dompdf/src/Positioner/Block.php] [Line 58]

### ISSUE THREAD

**Issue by nickpoulos** [barryvdh/laravel-dompdf#377]
vendor/dompdf/dompdf/src/Positioner/Block.php Line 58 -  

$y += $top;   

is failing. I am trying to play with DOMPDF versions to see if that helps. Not sure if this is truly a dompdf issue or this project's issue.

```
// Relative positionning
        if ($style->position === "relative") {
            $top = $style->length_in_pt($style->top, $cb["h"]);
            //$right =  $style->length_in_pt($style->right,  $cb["w"]);
            //$bottom = $style->length_in_pt($style->bottom, $cb["h"]);
            $left = $style->length_in_pt($style->left, $cb["w"]);
 
            $x += $left;
            $y += $top;
        }
```

**Comment by blyleven** [comment]
I'm having the same issue too. 

A non-numeric value encountered

array('frame' => object(Block), 'style' => object(Style), 'cb' => array('x' => 37.5, 'y' => 301.5, 'w' => 528.75, 'h' => 792.0, 37.5, 301.5, 528.75, 792.0), 'p' => object(TableCell), 'float' => 'none', 'y' => 301.5, 'x' => 37.5, 'top' => 'auto', 'left' => 0.0))

in Block.php (line 58)

**Comment by ashandi** [comment]
I had this issue too. The problem solved by @blyleven way, thanks.

**Comment by realripley00** [comment]
I was having this same non-numeric value encountered error that everyone keeps running into.  Specifically mine said "ErrorException in Style.php line 1326: A non-numeric value encountered" whenever I tried to download any generated PHP. 

I'm using version 8.1 (and 8.0 of the base  Dompdf/dompdf package, ). I'm obviously also using PHP 7, 7.1.1 specifically, and for the last two days I kept getting that same issue!

I did eventually get it to work and I wanted throw out my solution in case anyone else is still having problems.

I updated two of the dependencies of the Dompdf/dompdf package.  Both of them were installed earlier in the project when I installed Laravel Cashier, so I imagine many of the people experiencing this problem may have installed that package beforehand.

I updated:
"name": "phenx/php-svg-lib",
"version": "0.4",

and 

"name": "phenx/php-font-lib",
"version": "0.5",

This doesn't seem to effect the local install Cashier package at all, but I haven't tried it on a production server.  Anyway, I hope that helps someone.  Great package. Thank you barryvdh.
